### PR TITLE
feat: implemented structured logging

### PR DESCRIPTION
**Pull request by skyerus**

### CODE REVIEW

**Review by james-milligan — APPROVED** [review]
Looks good, nice work

File: README.md [hunk 1]
```diff
@@ -28,11 +28,35 @@ import (
...
+If not configured, the logger falls back to the standard Go log package at error level only.
+
+In order to avoid coupling to any particular logging implementation the sdk uses the structured logging [logr](https://github.com/go-logr/logr)
+API. This allows integration to any package that implements the layer between their logger and this API.
+Thankfully there is already [integration implementations](https://github.com/go-logr/logr#implementations-non-exhaustive)
+for many of the popular logger packages.
+
+```go
+var l logr.Logger
+l = integratedlogr.New() // replace with your chose integrator
```
Comment (toddbaert) [inline README.md:49]:
```suggestion
l = integratedlogr.New() // replace with your chosen integrator
```

File: pkg/openfeature/logger.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// logs using the standard log package on error, all other logs are no-ops
+type logger struct{}
+
+func (l logger) Init(info logr.RuntimeInfo) {}
+
+func (l logger) Enabled(level int) bool { return true }
+
+func (l logger) Info(level int, msg string, keysAndValues ...interface{}) {}
+
+func (l logger) Error(err error, msg string, keysAndValues ...interface{}) { log.Println(err) }
```
Comment (toddbaert) [inline pkg/openfeature/logger.go:24]: Is there a newline missing here?

**Review by toddbaert — APPROVED** [review]
LGTM, minor suggestions.

One thing I'm wondering is how to log in hooks. It's not immediately obvious to me if that's possible.

**Comment by skyerus** [comment]
> LGTM, minor suggestions.
> 
> One thing I'm wondering is how to log in hooks. It's not immediately obvious to me if that's possible.

That would be something for the application author to decide, there's nothing stopping them from placing a logger inside their hook struct and using it in the callbacks.

**Pull request merged**